netbox/users/views.py

```diff
@@ -36,6 +36,15 @@ def dispatch(self, *args, **kwargs):
         return super().dispatch(*args, **kwargs)
 
     def get(self, request):
+        if request.user.is_authenticated:
+            # Already logged-in, determine where to redirect
+            redirect_to = request.GET.get('next', reverse('home'))
+            if redirect_to and not is_safe_url(url=redirect_to, allowed_hosts=request.get_host()):
+                logger.warning(f"Ignoring unsafe 'next' URL passed to login form: {redirect_to}")
+                redirect_to = reverse('home')
```

Comment:
This duplicates code within `post()` below. This logic should be moved into a separate method that can be used for both requests.

Reply:
I've moved the code to an own method, please recheck.